askama_shared/src/generator.rs

```diff
@@ -1329,16 +1319,27 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
                     _ => return Err("loop.cycle(…) expects exactly one argument".into()),
                 },
                 s => return Err(format!("unknown loop method: {:?}", s).into()),
+            },
+            left => {
+                match *left {
```

Comment:
This seems like something that could easily be forgotten, to be updated, when changing the parser. Could it maybe be "simplified" or handled in the parser, by having some `enum CallKind`, which is a third value of `Expr::Call()`? :)

Reply:
Please see my comment in #590: https://github.com/djc/askama/pull/590#discussion_r779486870. I think the list of special cases is exhaustive. I think not even the `Expr::Unary(..) | Expr::BinOp(..) | Expr::Range(..)` case is actually need, because the parser won't ever generate an `Expr::Call(Expr::Unary(..), ..)`, only `Call(Group(Unary(..)), ..)`.